ai_genomics/utils/patent_data/get_ai_genomics_patents_utils.py

```diff
@@ -0,0 +1,84 @@
+"""Script to instantiate Google BigQuery client and
+functions to create bespoke genomics patents Google BigQuery
+tables in local project.
+"""
+######################################
+from google.cloud import bigquery
+from google.oauth2 import service_account
+import os
+
+######################################
+
+
+def est_conn():
+    """Instantiate Google BigQuery client
+    to query patent data.
+    """
+    if "GOOGLE_APPLICATION_CREDENTIALS" in os.environ:
+        google_creds = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+
+        credentials = service_account.Credentials.from_service_account_file(
+            google_creds
+        )
+
+        client = bigquery.Client(credentials=credentials)
+
+        return client
+
+    else:
+        print(
+            "export GOOGLE_APPLICATION_CREDENTIALS directory path as global variable."
+        )
+
+
+def clean_ipc_codes(genomics_codes: dict) -> list:
+    """Prepares IPC codes in format to query Google BigQuery.
+
+    Inputs:
+        genomics_codes (dict): Dictionary of patent classification codes relevant to genomics.
+    Outputs:
+        ipc_codes_clean (list): List of clean IPC codes relevant to genomics.
+    """
+    ipc_codes_clean = []
+
+    if "ipc" in genomics_codes.keys():
+        for ipc_code in genomics_codes["ipc"].keys():
+            if len(ipc_code) == 14:
+                if ipc_code[6] == "0":
+                    ipc_codes_clean.append(
+                        ipc_code[:4] + ipc_code[7:8] + "/" + ipc_code[8:10]
+                    )
+                else:
+                    ipc_codes_clean.append(
+                        ipc_code[:4] + ipc_code[6:8] + "/" + ipc_code[8:10]
+                    )
+            else:
+                ipc_codes_clean.append(ipc_code + "/")
```

Comment:
Would consider pulling this out into another function that has a docstring which explains why these indices are used.